File: 2018/solutions/E_cycle_ab.py
f = open("cycle.in", "r") 
n, v = map(int, f.readline().split())
v = v - 1
s = [f.readline() for i in range(0, n)]
res = 0
for a in range(0, n):
    for b in range(0, n):
        if (s[v][a] == s[a][b] == s[b][v] == '1'):
            res = res + 1

f = open("cycle.out", "w")
if res == 0:
    print("-1\n", file=f)
else:
    print("3 " + str(res) + "\n", file=f)

File: 2018/solutions/E_cycle_ab.cpp
#include<iostream>
using namespace std;

const int maxN = 1000 + 10;

char s[maxN][maxN];

int main() {
    freopen("cycle.in", "r", stdin);
    freopen("cycle.out", "w", stdout);
    int n, v;
    scanf("%d%d", &n, &v);
    v--;
    for(int i = 0; i < n; i++) 
        scanf("%s", s[i]);

    int res = 0;
    for(int a = 0; a < n; a++)
        for(int b = 0; b < n; b++)
            if(s[v][a] == '1' && s[a][b] == '1' && s[b][v] == '1') {
                res++;
            }

    if(res != 0)
        printf("3 %d\n", res);
    else
        printf("-1\n");
    return 0;
}
